CRM/Core/Error.php

```diff
@@ -466,17 +466,12 @@ function_exists($config->fatalErrorHandler)
     $template = CRM_Core_Smarty::singleton();
     $template->assign($vars);
     $content = $template->fetch('CRM/common/fatal.tpl');
+
     if ($config->backtrace) {
       $content = self::formatHtmlException($exception) . $content;
     }
-    if ($config->userFramework == 'Joomla' &&
-      class_exists('JError')
-    ) {
-      JError::raiseError('CiviCRM-001', $content);
-    }
-    else {
-      echo CRM_Utils_System::theme($content);
-    }
```

Comment:
@vingle wrote:

> anyone currently using custom styled, front-end error pages that match their public-facing theme would, I guess, lose that with this commit.

`civicrm-core` has a setting `fatalErrorHandler` which you can use to define a custom function for emitting an error. I wonder if it work as an off-ramp to:

* Put this old code-block in a separate function (e.g. `CRM_Utils_System_Joomla::mapToJError` or `CRM_Core_Error::useJoomlaError`)
* In the post-upgrade message on Joomla sites, advise that the error-handling has been normalized to consistently display error messages - but that you can continue using `JError` by setting the option `fatalErrorHandler` to `CRM_Utils_System_Joomla::mapToJError` (or whatever it's called)

(*Of course, that would be have to be tested, and I don't know off the top of my head the best place to put that function..*)

OTOH, if no one is likely to use the option, then just KISS and carry on without...